### PR TITLE
feat(rating-menu): implement `getRenderState` and `getWidgetRenderState`

### PR DESCRIPTION
**Pull request by yannickcr**
This implements the `getRenderState` and `getWidgetRenderState` widget lifecycle hooks in `ratingMenu`.

### CODE REVIEW

File: src/connectors/rating-menu/__tests__/connectRatingMenu-test.js [hunk 1]
```diff
@@ -419,6 +424,184 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/rating-menu
     });
   });
 
+  describe('getRenderState', () => {
+    it('returns the render state', () => {
+      const renderFn = jest.fn();
+      const unmountFn = jest.fn();
+      const createRatingMenu = connectRatingMenu(renderFn, unmountFn);
+      const rangeWidget = createRatingMenu({
```
Comment (Haroenv) [inline src/connectors/rating-menu/__tests__/connectRatingMenu-test.js:432]: you have `rangeWidget` instead of ratingWidget a few times

**Review by eunjae-lee — APPROVED** [review]
looks good to me

**Review by Haroenv — APPROVED** [review]
looks good! (technically a breaking change, but not on public api so it's alright for me)

**Review by Haroenv — CHANGES_REQUESTED** [review]
the breaking change you did requires you update the rating menu tests: https://app.circleci.com/pipelines/github/algolia/instantsearch.js/4950/workflows/50783eb6-0218-4c6b-915d-745f888967c5/jobs/21823

**Comment by yannickcr** [comment]
Fixed the tests, and yes since it is not a public API this change is fine IMHO :slightly_smiling_face:

**Review by devdreamsolution — APPROVED** [review]

**Review by Haroenv — APPROVED** [review]
Good enough for now, but the getWidgetRenderState & getWidgetState tests could be split up

**Pull request merged**